### PR TITLE
Add psycopg3 compat

### PR DESCRIPTION
**Pull request by aleksandr-shtaub**
A naive implementation of connector for psycopg3. 
Basically, it's just a modified copy of current aiopg connector.

### Successful PR Checklist:

- [x] Tests
- [x] Documentation

### CODE REVIEW

File: procrastinate/psycopg3_connector.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+                    continue
+                raise exc
+        raise exceptions.ConnectorException(
+            f"Could not get a valid connection after {max_tries} tries"
+        ) from final_exc
+
+    return wrapped
+
+
+PERCENT_PATTERN = re.compile(r"%(?![\(s])")
```
Comment (aleksandr-shtaub) [inline procrastinate/psycopg3_connector.py:80]: Unused. Not sure if it's needed

**Comment by ewjoachim** [comment]
Hello :)

Ha, I had started doing the same thing on my side in an attempt to solve the sync/async issues (see https://github.com/procrastinate-org/procrastinate/pull/753/commits)

I'm really thinking about doing a v1 and dropping all other connectors except psycopg3.

It seems everyone is headed the same way :) No idea yet what the right approach would be though.

**Comment by aleksandr-shtaub** [comment]
Ah, didn't see it, my bad :) 
Searching by "psycopg3" in GitHub doesn't show this.
Anyways, it was a good exercise to learn the differences between aiopg+psycopg2 and psycopg3.

**Comment by ewjoachim** [comment]
You couldn't have seen it, I had committed it but not pushed yet (I have after I saw your PR)

Here's the diff between our 2 connectors: 
https://www.diffchecker.com/V89PnXHK/

**Comment by ewjoachim** [comment]
I think I'm leaning towards working with your PR first, maybe backporting some of my changes, and then do the big bang where I fix sync/async and remove other connectors in a different PR.

**Comment by aleksandr-shtaub** [comment]
Okay, as you prefer. 
Feel free to change the PR. I'll probably be busy until the next weekend.

**Review by ewjoachim — APPROVED** [review]

**Comment by ewjoachim** [comment]
Given that it's (supposedly) not breaking anything, I'll merge and release. Feel free to report on issues you findonce you start using it.

**Comment by ewjoachim** [comment]
https://pypi.org/project/procrastinate/0.31.0/

**Pull request merged**